rct229/rules/section23/section23rule2.py

```diff
@@ -0,0 +1,91 @@
+from rct229.data.schema_enums import schema_enums
+from rct229.rule_engine.rule_base import RuleDefinitionBase
+from rct229.rule_engine.rule_list_indexed_base import RuleDefinitionListIndexedBase
+from rct229.rule_engine.user_baseline_proposed_vals import UserBaselineProposedVals
+from rct229.ruleset_functions.baseline_system_type_compare import (
+    baseline_system_type_compare,
+)
+from rct229.ruleset_functions.baseline_systems.baseline_system_util import HVAC_SYS
+from rct229.ruleset_functions.get_baseline_system_types import get_baseline_system_types
+from rct229.schema.config import ureg
+from rct229.utils.pint_utils import CalcQ
+from rct229.utils.std_comparisons import std_equal
+
+APPLICABLE_SYS_TYPES = [
+    HVAC_SYS.SYS_5,
+    HVAC_SYS.SYS_6,
+    HVAC_SYS.SYS_7,
+    HVAC_SYS.SYS_8,
+    HVAC_SYS.SYS_11_1,
+]
+
+FanSystemTemperatureControl = schema_enums["FanSystemTemperatureControlOptions"]
+REQUIRED_RESET_DIFF_TEMP = 5.0 * ureg("degR")
+
+
+class Section23Rule2(RuleDefinitionListIndexedBase):
+    """Rule 2 of ASHRAE 90.1-2019 Appendix G Section 23 (Air-side)"""
+
+    def __init__(self):
+        super(Section23Rule2, self).__init__(
+            rmrs_used=UserBaselineProposedVals(False, True, False),
+            each_rule=Section23Rule2.HVACRule(),
+            index_rmr="baseline",
+            id="23-2",
+            description="For baseline systems 5-8 and 11, the SAT is reset higher by 5F under minimum cooling load conditions.",
+            ruleset_section_title="HVAC - Airside",
+            standard_section="Section G3.1.3.12 Supply Air Temperature Reset (Systems 5 through 8 and 11)",
+            is_primary_rule=True,
+            rmr_context="ruleset_model_instances/0",
+            list_path="$..heating_ventilating_air_conditioning_systems[*]",
+        )
+
+    def is_applicable(self, context, data=None):
+        rmi_b = context.baseline
+        baseline_system_types_dict = get_baseline_system_types(rmi_b)
+
+        return any(
+            [
+                baseline_system_type_compare(system_type, applicable_sys_type, False)
+                for system_type in baseline_system_types_dict.keys()
+                for applicable_sys_type in APPLICABLE_SYS_TYPES
+            ]
+        )
+
+    class HVACRule(RuleDefinitionBase):
+        def __init__(self):
+            super(Section23Rule2.HVACRule, self).__init__(
+                rmrs_used=UserBaselineProposedVals(False, True, False),
+                required_fields={
+                    "$": ["fan_system"],
+                    "fan_system": [
+                        "temperature_control",
+                        "reset_differential_temperature",
+                    ],
+                },
+            )
+
+        def get_calc_vals(self, context, data=None):
+            fan_system_b = context.baseline["fan_system"]
```

Comment:
I suggest that we always create a variable for the context itself to document what the context is at this level.
```
hvac_b = context.baseline
fan_system_b = hvac_b["fan_system"]
...
```

Reply:
Good point! Fixed.